lib/zipkin-tracer/rack/zipkin-tracer.rb

```diff
@@ -52,32 +51,20 @@ def annotate_plugin(env, status, response_headers, response_body)
     end
 
     def trace!(span, zipkin_env, &block)
-      synchronize do
```

Comment:
We do not need synchronize, we are using local vars now. It gives us a modest improvement (do not need to initialize and hold a semaphore)